modules/color-picker/react/spec/ColorInput.spec.tsx

```diff
@@ -1,115 +1,140 @@
-    );
-
-    const event = {
-      currentTarget: {value: 'e6e'},
-    };
-    component.prop('onChange')(event);
-    expect(onChangeMock).toBeCalledWith(event);
-  });
-  test('should call onChange prop onChange with stripped hash', () => {
-    const onChangeMock = jest.fn();
-    const component = mount(
-      <ColorInput onChange={onChangeMock} onValidColorChange={jest.fn()} value={''} />
-    );
-    component.setProps({value: '#e6e'});
-
-    expect(component.find('input').props().value).toBe('e6e');
-  });
-  test('should call onValidColorChange prop onChange with valid hex value', () => {
-    const onValidColorChangeMock = jest.fn();
-    const component = mount(
-      <ColorInput onChange={jest.fn()} onValidColorChange={onValidColorChangeMock} value={'#e6e'} />
-    );
-    const event = {
-      target: {value: '#e6e'},
-    };
-    const input = component.find('input');
-    input.simulate('change', event);
-    expect(onValidColorChangeMock).toBeCalledWith('#ee66ee');
-  });
-  test('should not call onValidColorChange prop onChange if hex code is not valid', () => {
-    const onValidColorChange = jest.fn();
-    const component = mount(
-      <ColorInput onChange={jest.fn()} onValidColorChange={onValidColorChange} value={''} />
-    );
-    const event = {
-      target: {value: 'e6ee'},
-    };
-    const input = component.find('input');
-    input.simulate('change', event);
-    expect(onValidColorChange).toHaveBeenCalledTimes(0);
-  });
-  test('should remove hash', () => {
-    const component = mount(
-      <ColorInput onChange={jest.fn()} onValidColorChange={jest.fn()} value={'#e6e'} />
-    );
-    expect(component.find('input').props().value).toBe('e6e');
-  });
-  test('should allow only 6 non-# characters as the value', () => {
-    const component = mount(
-      <ColorInput onChange={jest.fn()} onValidColorChange={jest.fn()} value={'#123456789'} />
-    );
-    expect(component.find('input').props().value).toBe('123456');
-  });
-  test('should show the checkmark after the valid hex value has been formatted', () => {
-    const component = mount(
-      <ColorInput onChange={jest.fn()} onValidColorChange={jest.fn()} value={'#123456789'} />
-    );
-    expect(
-      component
-        .find('div[className^="css"]')
-        .at(1) // This div the background of the color preview/checkmark bit
-        .prop('style')
-    ).toHaveProperty('backgroundColor', '#123456');
-  });
-  test('should set inputRef as reference to native input', () => {
-    const ref = React.createRef<HTMLInputElement>();
-    mount(<ColorInput inputRef={ref} value={''} />);
-    expect(ref.current && ref.current.tagName && ref.current.tagName.toUpperCase()).toBe('INPUT');
-  });
+  describe('when rendered', () => {
+    describe('with a placeholder', () => {
+      test('should render a ColorInput with a placeholder', () => {
+        const {getByTestId} = render(
```

Comment:
Based on the [recommendations in the docs](https://testing-library.com/docs/dom-testing-library/api-queries#bytestid), we should avoid `getByTestId` wherever possible. I think you'll be most successful with `getByRole('textbox')` here so it doesn't require any extra props.

Reply:
Will do!